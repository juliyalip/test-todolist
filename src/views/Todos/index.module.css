.textNotFound {
  width: 100%;
  background-color: rgb(212, 208, 208);
  text-align: center;
  padding: 10px;
  border-radius: 4px;
}/*# sourceMappingURL=index.module.css.map */