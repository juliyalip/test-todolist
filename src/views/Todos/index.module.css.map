{"version":3,"sources":["index.module.scss","index.module.css"],"names":[],"mappings":"AAAA;EACI,WAAA;EACA,oCAAA;EACA,kBAAA;EACA,aAAA;EACA,kBAAA;ACCJ","file":"index.module.css"}