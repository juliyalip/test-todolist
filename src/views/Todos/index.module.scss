.textNotFound{
    width: 450px;
    background-color: rgb(212, 208, 208);
    text-align: center;
    padding: 10px ;
    border-radius: 4px;
}