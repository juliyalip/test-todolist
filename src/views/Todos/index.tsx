import { useAppSelector } from 'redux/hooks';
import { getTodos } from 'redux/selectors';
import TodoForm from 'components/TodoForm';
import TodoList from 'components/TodoList';
import Hero from 'components/Hero';
import ContainerForPosition from 'components/itoms/ContainerForPosition';
import ControllingSection from 'components/ControllingSection';
import Filter from 'components/Filter';
import styles from './index.module.scss';

export default function Todos() {
  const todos = useAppSelector(getTodos);
  const isNotData = Boolean(todos.length === 0);
  return (
    <>
      <Hero />

      <TodoForm />
      <ContainerForPosition>
        {isNotData && <p className={styles.textNotFound} >Data not found</p>}
        {todos.length > 0 && <Filter />}
        {todos && <TodoList />}
        {todos.length > 0 && <ControllingSection />}
      </ContainerForPosition>
    </>
  );
}
