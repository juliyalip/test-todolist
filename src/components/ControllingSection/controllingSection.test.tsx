import { fireEvent} from '@testing-library/react';
import Todos from 'views/Todos';
import { renderWithProviders } from 'utils/test-utils';

const initionTodos = [
  { id: '001', text: 'I like React', done: true },
  { id: '002', text: 'I like Redux', done: true },
];

const renderComponent = () =>
  renderWithProviders(<Todos />, {
    preloadedState: {
      todos: initionTodos,
    },
  });

describe('test controling section', () => {
  it('Pre render correctly', () => {
    const { getByText } = renderComponent();
    
    expect(getByText(/I like React/)).toBeInTheDocument();
    expect(getByText(/I like Redux/)).toBeInTheDocument();
    expect(getByText(/Clear complited/)).toBeInTheDocument();
    expect(getByText(/Active/i)).toBeInTheDocument();
    expect(getByText(/All/i)).toBeInTheDocument();
    expect(getByText(/Completed/i)).toBeInTheDocument();

    fireEvent.click(getByText('Completed'));
  });

  it("click button 'completed'", () => {
    const { getByText, getByTestId } = renderComponent();
    const result = getByTestId('result');
    fireEvent.click(getByText('Completed'));

    expect(result.textContent).toBe('2 items');
  });

  it("click on button 'active'", () => {
    const { getByText, getByTestId } = renderComponent();
    const result = getByTestId('result');
    fireEvent.click(getByText(/Active/i));

    expect(result.textContent).toBe('0 items');
  });

  it("click on button 'all'", () => {
    const { getByText, getByTestId } = renderComponent();
    const result = getByTestId('result');
    fireEvent.click(getByText(/All/i));

    expect(result.textContent).toBe('2 items');
  });

  it('get result after delete all complited todos', () => {
    const { getByText, queryByText } = renderComponent();

    expect(queryByText(/Data not found/i)).toBeNull();
    
    fireEvent.click(getByText(/Clear complited/i));

    expect(getByText(/Data not found/i)).toBeInTheDocument();
    expect(getByText(/Data not found/i)).toMatchSnapshot();
  });
});
