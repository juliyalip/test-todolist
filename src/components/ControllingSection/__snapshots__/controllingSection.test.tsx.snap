// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test controling section get result after delete all complited todos 1`] = `
<p
  class="textNotFound"
>
  Data not found
</p>
`;
